wazo_confd/plugins/line/schema.py

```diff
@@ -29,9 +31,55 @@ class LineSchema(BaseSchema):
     application = Nested(
         'ApplicationSchema', only=['uuid', 'name', 'links'], dump_only=True
     )
+    endpoint_sip = Nested('EndpointSIPSchema')
+    endpoint_sccp = Nested('SccpSchema')
+    endpoint_custom = Nested('CustomSchema')
+    extensions = Nested('ExtensionSchema', many=True)
+    users = Nested(
+        'UserSchema',
+        only=['uuid', 'firstname', 'lastname', 'links'],
+        many=True,
+        dump_only=True,
+    )
+
+    @validates_schema
+    def _validate_only_one_endpoint(self, data, **kwargs):
+        nb_endpoint = 0
+        if data.get('endpoint_sip'):
+            nb_endpoint += 1
+        if data.get('endpoint_sccp'):
+            nb_endpoint += 1
+        if data.get('endpoint_custom'):
+            nb_endpoint += 1
+
+        if nb_endpoint > 1:
+            raise ValidationError('Only one endpoint should be configured')
+
+    @pre_load
+    def populate_missing_context(self, data, **kwargs):
+        extensions = data.get('extensions', [])
+        for ext in extensions:
+            if 'context' not in ext:
+                ext['context'] = data.get('context')
+        return data
+
+
+class LineSchemaNullable(LineSchema):
+    def on_bind_field(self, field_name, field_obj):
+        super().on_bind_field(field_name, field_obj)
+        nullable_fields = ['provisioning_code', 'position', 'registrar']
+        if field_name in nullable_fields:
+            field_obj.allow_none = True
+
+
+class LineListSchema(LineSchemaNullable):
```

Comment:
I may miss something, but I don't understand why you need to inherit from Nullable schema here?
LineListSchema is (or should) only be used with `dump` but allowing none is a restriction when we do `load`